src/main/java/com/example/bookstore/controller/BookController.java

```diff
@@ -30,6 +30,7 @@ public class BookController {
     private final BookService bookService;
     private final BookMapper bookMapper;
 
+    @PreAuthorize("hasRole('ROLE_USER, ROLE_ADMIN')")
```

Comment:
are you sure it's correct specification of roles?
maybe it should be something like `hasAnyRole('ROLE_USER', 'ROLE_ADMIN')`